common/lib/xmodule/xmodule/modulestore/django.py

```diff
@@ -243,9 +244,38 @@ class ModuleI18nService(object):
     i18n service.
 
     """
-
-    def __getattr__(self, name):
-        return getattr(django.utils.translation, name)
+    def ugettext(self, string, **kwargs):
+        """
+        This operation is a proxy for django.utils.translation.ugettext, which is itself a proxy for
+        the GNU gettext ugettext operation.  Here we attempts to look up the provided string in the
+        XBlock's own domain translation catalog, currently expected to be:
+            <xblock_root>/conf/locale/<language>/LC_MESSAGES/<domain>.po
+        If ModuleI18nService can't locate the domain translation catalog then we fall-back onto
+        django.utils.translation.ugettext, which will attempt to find a matching string in the
+        LMS' own domain translation catalog -- effectively achieving translation by coincidence.
+        """
+        # The translation workflow should only execute if there's an actual string to look up
+        # If gettext processes an empty string the PO file header information will oddly be returned
+        translated_string = unicode(string)
+        if translated_string:
+            try:
+                # TODO: Move these values into settings.  We might allow for multiple possibilities
+                # for these values through the use of lists or tuples.  For example, instead of
+                # 'django' an XBlock developer could specify 'xblock_name' for their PO file.
+                xblock_domain = 'django'
+                xblock_locale = 'conf/locale'
+                locale_path = kwargs.get('xblock_root', '') + xblock_locale
+                translator = gettext.translation(
+                    xblock_domain,
+                    locale_path,
+                    [to_locale(get_language())]
+                )
+                _ = translator.ugettext
+            except:
```

Comment:
It would be good if we can avoid a bare except here -- I think there's a specific exception raised when the domain/catalog is not found?